app/proposal.go

```diff
@@ -95,33 +112,13 @@ func (h *ProposalHandler) ValidateTransaction(tx sdk.Tx) error {
 	return nil
 }
 
-func (h *ProposalHandler) PrepareProposalHandler() sdk.PrepareProposalHandler {
-	return func(ctx sdk.Context, req abci.RequestPrepareProposal) abci.ResponsePrepareProposal {
-		var maxBlockGas uint64
-		if b := ctx.ConsensusParams().Block; b != nil {
-			maxBlockGas = uint64(b.MaxGas)
-		}
-
-		defer h.TxSelector.Clear()
-
-		for _, txBz := range req.Txs {
-			memTx, err := h.TxDecoder(txBz)
-			if err != nil {
-				continue
-			}
-
-			if err := h.ValidateTransaction(memTx); err != nil {
-				continue
-			}
-
-			stop := h.TxSelector.SelectTxForProposal(uint64(req.MaxTxBytes), maxBlockGas, memTx, txBz)
-			if stop {
-				break
-			}
-		}
-
-		return abci.ResponsePrepareProposal{Txs: h.TxSelector.SelectedTxs()}
-	}
+func (h *ProposalHandler) PrepareProposalHandler(txVerifier baseapp.ProposalTxVerifier) sdk.PrepareProposalHandler {
+	defaultHandler := baseapp.NewDefaultProposalHandler(h.mempool, txVerifier)
```

Comment:
you want to create a nested handler, right? why not construct the parent handler once in constructor?

Reply:
yea, but no difference to pass default handler in, both construct once right?

Reply:
yes, it's called once as well.
how about the `ProcessProposalHandler`?

Reply:
yes! haven't try process before

Reply:
I did another round of cleanup